crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -155,6 +154,13 @@ impl Default for TimelineInnerSettings {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+enum TimelineFocusKind {
```

Comment:
Reusing the `TimelineFocus` is an interesting idea: i tried that, but it required either adding `Clone` to it (which would require in turn many other structs to be `Clone`able, including the non-trivial Paginator one), or moving the `focus_data` into the `TimelineInnerState`, which I didn't want at the time (because holding onto the focus data is needed for paginating, and I didn't want to block the timeline on pagination requests).

So I'll keep it like this for now, but good suggestion :grinning: